mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/quota/InMemoryCurrentQuotaManager.java

```diff
@@ -41,14 +40,14 @@
 
 public class InMemoryCurrentQuotaManager implements StoreCurrentQuotaManager {
 
-    private final LoadingCache<QuotaRoot, Entry> quotaCache;
+    private final LoadingCache<QuotaRoot, CurrentQuotas> quotaCache;
```

Comment:
LoadingCache doesn't provide CAS semantic so you need an AtomicReference as value.

Reply:
This is loosely correlated with the work @Arsnael did here.

Can't it be handled in another pull request?

Reply:
I can try to check I guess...

Reply:
@mbaechler : let me know if my last fixup answers your concern !